server/graphql/v1/mutations/orders.js

```diff
@@ -528,39 +527,80 @@ export async function createOrder(order, loaders, remoteUser, reqIp) {
       collective.addUserWithRole({ id: user.id, CollectiveId: order.ReferralCollectiveId }, roles.FUNDRAISER);
     }
 
-    // Share suspicious transactions on Slack
-    if (recaptchaResponse && recaptchaResponse.score && recaptchaResponse.score <= 0.5) {
-      slackLib
-        .postActivityOnPublicChannel(
-          {
-            type: activities.ORDERS_SUSPICIOUS,
-            data: {
-              order,
-              user,
-              fromCollective,
-              collective,
-              recaptchaResponse,
-            },
-          },
-          config.slack.webhookUrl,
-          {
-            channel: config.slack.abuseChannel,
-          },
-        )
-        .catch(console.log);
-    }
-
     return order;
   } catch (error) {
-    debug('createOrder mutation error: ', error);
-    if (orderCreated && !orderCreated.processedAt) {
-      // TODO: Order should be updated with data JSON field to store the error to review later
-      orderCreated.update({ status: status.ERROR });
+    console.log(error);
+
+    if (orderCreated) {
+      if (!orderCreated.processedAt) {
+        // TODO: Order should be updated with data JSON field to store the error to review later
+        orderCreated.update({ status: status.ERROR });
+      }
+
+      if (!error.stripeResponse) {
+        throw error;
+      }
+
+      orderCreated.stripeError = {
+        message: error.message,
+        account: error.stripeAccount,
+        response: error.stripeResponse,
+      };
+
+      return orderCreated;
     }
+
     throw error;
   }
 }
 
+export async function confirmOrder(order, remoteUser) {
+  if (!remoteUser) {
+    throw new errors.Unauthorized({ message: 'You need to be logged in to confirm an order' });
+  }
+
+  const existingOrder = await models.Order.findOne({
+    where: {
+      id: order.id,
+    },
+    include: [
+      { model: models.Collective, as: 'collective' },
+      { model: models.Collective, as: 'fromCollective' },
+      { model: models.PaymentMethod, as: 'paymentMethod' },
+    ],
+  });
+
+  if (!existingOrder) {
+    throw new errors.NotFound({ message: 'Order not found' });
+  }
+  if (!remoteUser.isAdmin(existingOrder.FromCollectiveId)) {
+    throw new errors.Unauthorized({ message: "You don't have permission to confirm this order" });
+  }
+  if (existingOrder.status !== 'ERROR' && existingOrder.status !== 'PENDING') {
+    throw new errors.NotFound({ message: 'Order can only be confirmed if ERROR or PENDING.' });
```

Comment:
`NotFound` doesn't feels to me like the right error type. Would be great to have a `ActionAlreadyDone` (or similar) error type for this case, I'd like to be able to catch it in the Frontend to show it as an info (`This order has alrady been confirmed`) rather than an error.

Reply:
Good that you spotted that. I absolutely didn't want to do that, copy/paste accident!